examples/keras_io/vision/keypoint_detection.py

```diff
@@ -0,0 +1,448 @@
+
+# Extract the colours and labels.
+colours = keypoint_def["Hex colour"].values.tolist()
+colours = ["#" + colour for colour in colours]
+labels = keypoint_def["Name"].values.tolist()
+
+
+# Utility for reading an image and for getting its annotations.
+def get_dog(name):
+    data = json_dict[name]
+    img_data = plt.imread(os.path.join(IMG_DIR, data["img_path"]))
+    # If the image is RGBA convert it to RGB.
+    if img_data.shape[-1] == 4:
+        img_data = img_data.astype(np.uint8)
+        img_data = Image.fromarray(img_data)
+        img_data = np.array(img_data.convert("RGB"))
+    data["img_data"] = img_data
+
+    return data
+
+
+"""
+## Visualize data
+
+Now, we write a utility function to visualize the images and their keypoints.
+"""
+
+
+# Parts of this code come from here:
+# https://github.com/benjiebob/StanfordExtra/blob/master/demo.ipynb
+def visualize_keypoints(images, keypoints):
+    fig, axes = plt.subplots(nrows=len(images), ncols=2, figsize=(16, 12))
+    [ax.axis("off") for ax in np.ravel(axes)]
+
+    for (ax_orig, ax_all), image, current_keypoint in zip(axes, images, keypoints):
+        ax_orig.imshow(image)
+        ax_all.imshow(image)
+
+        # If the keypoints were formed by `imgaug` then the coordinates need
+        # to be iterated differently.
+        if isinstance(current_keypoint, KeypointsOnImage):
+            for idx, kp in enumerate(current_keypoint.keypoints):
+                ax_all.scatter(
+                    [kp.x], [kp.y], c=colours[idx], marker="x", s=50, linewidths=5
+                )
+        else:
+            current_keypoint = np.array(current_keypoint)
+            # Since the last entry is the visibility flag, we discard it.
+            current_keypoint = current_keypoint[:, :2]
+            for idx, (x, y) in enumerate(current_keypoint):
+                ax_all.scatter([x], [y], c=colours[idx], marker="x", s=50, linewidths=5)
+
+    plt.tight_layout(pad=2.0)
+    plt.show()
+
+
+# Select four samples randomly for visualization.
+samples = list(json_dict.keys())
+num_samples = 4
+selected_samples = np.random.choice(samples, num_samples, replace=False)
+
+images, keypoints = [], []
+
+for sample in selected_samples:
+    data = get_dog(sample)
+    image = data["img_data"]
+    keypoint = data["joints"]
+
+    images.append(image)
+    keypoints.append(keypoint)
+
+visualize_keypoints(images, keypoints)
+
+"""
+The plots show that we have images of non-uniform sizes, which is expected in most
+real-world scenarios. However, if we resize these images to have a uniform shape (for
+instance (224 x 224)) their ground-truth annotations will also be affected. The same
+applies if we apply any geometric transformation (horizontal flip, for e.g.) to an image.
+Fortunately, `imgaug` provides utilities that can handle this issue.
+In the next section, we will write a data generator inheriting the
+[`keras.utils.Sequence`](https://keras.io/api/utils/python_utils/#sequence-class) class
```

Comment:
Convert `Sequence` to `PyDataset`. For performance you probably want to set `workers`.